Snabble/UI/Payment/QRCheckoutViewController.swift

```diff
@@ -22,29 +22,110 @@ final class QRCheckoutViewController: UIViewController {
+        qrCodeView.translatesAutoresizingMaskIntoConstraints = false
+        qrCodeView.contentMode = .scaleToFill
+        qrCodeView.setContentHuggingPriority(.defaultLow + 1, for: .horizontal)
+        qrCodeView.setContentHuggingPriority(.defaultLow + 1, for: .vertical)
+
+        let explanationBottomLabel = customLabel
+        explanationUpperLabel.font = UIFont.systemFont(ofSize: 11)
+        explanationBottomLabel.numberOfLines = 0
+
+        let cancelButton = UIButton(type: .system)
+        cancelButton.translatesAutoresizingMaskIntoConstraints = false
+        cancelButton.setTitle(L10n.Snabble.cancel, for: .normal)
+        cancelButton.titleLabel?.font = UIFont.systemFont(ofSize: 15)
+        cancelButton.setTitleColor(SnabbleUI.appearance.accentColor, for: .normal)
+        cancelButton.isEnabled = true
+        cancelButton.isUserInteractionEnabled = true
+        cancelButton.addTarget(self, action: #selector(cancelButtonTapped(_:)), for: .touchUpInside)
+
+        contentView.addSubview(checkoutIdLabel)
+        contentView.addSubview(totalPriceLabel)
+        contentView.addSubview(explanationUpperLabel)
+        contentView.addSubview(qrCodeView)
+        contentView.addSubview(explanationBottomLabel)
+        contentView.addSubview(cancelButton)
+
+        NSLayoutConstraint.activate([
+            checkoutIdLabel.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 8),
+            checkoutIdLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            checkoutIdLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+
+            qrCodeView.centerXAnchor.constraint(equalTo: contentView.centerXAnchor),
+            qrCodeView.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
+            qrCodeView.heightAnchor.constraint(equalTo: qrCodeView.widthAnchor),
+
+            explanationUpperLabel.bottomAnchor.constraint(equalTo: qrCodeView.topAnchor, constant: -16),
+            explanationUpperLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            explanationUpperLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+
+            totalPriceLabel.bottomAnchor.constraint(equalTo: explanationUpperLabel.topAnchor, constant: -16),
+            totalPriceLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            totalPriceLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+
+            explanationBottomLabel.topAnchor.constraint(equalTo: qrCodeView.bottomAnchor, constant: 16),
+            explanationBottomLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16),
+            explanationBottomLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16),
+
+            cancelButton.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -20),
```

Comment:
Cancel Button is too low. It should be constrainted to the safeAreaLayoutGuide.

Reply:
Fixed